packages/hyperion-autologging/src/ALUIEventPublisher.ts

```diff
@@ -15,14 +15,15 @@ import { ReactComponentData } from "./ALReactUtils";
 import { getSurfacePath } from "./ALSurfaceUtils";
 import { ALFlowletEvent, ALReactElementEvent, ALSharedInitOptions, ALTimedEvent } from "./ALType";
 
-export type ALUIEvent = Readonly<{
-  domEvent: GlobalEventHandlersEventMap[ALUIEvent['event']],
-  event: keyof GlobalEventHandlersEventMap,
-  element: Element | null,
-  elementName?: string | null,
-  isTrusted: boolean,
-
-}>;
+export type ALUIEvent<T = EventHandlerMap> = {
```

Comment:
Took me a while to understand this is not a nested object. May be either add some more comments, or break this into a `ALUIEventMap` and then define `ALUIEvent` accordingly. 

Also, I'm not sure you need to or should export this as a generic type. We don't expect external modules to change these types.